src/main/java/com/demo/top/itunes/ItunesGateway.java

```diff
@@ -0,0 +1,70 @@
+package com.demo.top.itunes;
+
+import com.demo.top.exception.FailedDependencyException;
+import com.demo.top.exception.InternalServerException;
+import com.demo.top.model.album.AlbumSearchResponse;
+import com.demo.top.model.artist.ArtistSearchResponse;
+import com.demo.top.model.song.SongSearchResponse;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Component;
+import org.springframework.web.client.RestTemplate;
+
+@Component
+public class ItunesGateway {
+
+  private RestTemplate restTemplate;
+
+  private String itunesUrl;
+
+  private ObjectMapper objectMapper;
+
+  @Autowired
+  public ItunesGateway(RestTemplate restTemplate,
+      ObjectMapper objectMapper) {
+    this.restTemplate = restTemplate;
+    this.objectMapper = objectMapper;
+  }
+
+  @Value("${top-application.itunes.url}")
```

Comment:
this can also be injected through constructor

Reply:
Done